tiledb/sm/array_schema/array_schema.cc

```diff
@@ -545,95 +586,150 @@ Status ArraySchema::drop_attribute(const std::string& attr_name) {
   return Status::Ok();
 }
 
-Status ArraySchema::deserialize(ConstBuffer* buff) {
+shared_ptr<ArraySchema> ArraySchema::deserialize(
+    ConstBuffer* buff, const URI& uri) {
+  Status st;
   // Load version
-  RETURN_NOT_OK(buff->read(&version_, sizeof(uint32_t)));
+  uint32_t version;
+  st = buff->read(&version, sizeof(uint32_t));
+  if (!st.ok())
+    throw std::runtime_error(
+        "[ArraySchema::deserialize] Error: Failed to load version.");
```

Comment:
This is probably not most important piece of data validation. If the format version is outside of a range that this code understands, it should refuse to create the object. For now, checking that it's the current version or less will be an adequate check.